src/screens/TopicPageScreen/index.js

```diff
@@ -387,6 +387,10 @@ const TopicPageScreen = (props) => {
     navigation.push('TopicMemberScreen', navigationParam);
   };
 
+  React.useEffect(() => {
+    onRefresh();
+  }, [topicId]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

While the `useEffect` hook is correctly used to refresh the feed when `topicId` changes, it's important to consider that `onRefresh` might be an asynchronous operation. If it involves network requests or other time-consuming tasks, it should be handled accordingly. 

To improve this, you can modify the code to handle `onRefresh` as an async function inside the `useEffect` hook.

```diff
-  React.useEffect(() => {
-    onRefresh();
-  }, [topicId]);
+  React.useEffect(() => {
+    const refresh = async () => {
+      await onRefresh();
+    };
+
+    refresh();
+  }, [topicId]);
```
This way, if `onRefresh` returns a promise (which is common for data fetching operations), it will be properly awaited.


<!-- This is an auto-generated comment by OSS CodeRabbit -->